cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreTableManager.java

```diff
@@ -524,6 +530,19 @@ private String generateFileSetCreateStatement(DatasetId datasetId, Dataset datas
     String databaseName = ExploreProperties.getExploreDatabaseName(properties);
     Map<String, String> tableProperties = FileSetProperties.getTableProperties(properties);
 
+    // if this dataset reuses an existing table, do not attempt to create it
+    if (FileSetProperties.isUseExisting(tableProperties)
+      || FileSetProperties.isPossessExisting(tableProperties)) {
+      try {
+        exploreService.getTableInfo(databaseName, tableName);
+        // table exists: do not attempt to create
+        return null;
+      } catch (Exception e) {
+        LOG.warn("Dataset '{}' is configured to use an existing explore table, but table '{}' does not " +
```

Comment:
Man, you catch everything :) 

The reason for this is that in truncate, we disable and reenable explore. If isPossessed is true, then the disable will drop the table, and the enable would not recreate it, instead it would fail saying that the table has to exist already. In FileSetAdmin, I was able to work around that by creating an internal method that gets a flag whether a truncate is in process. In Explore I would have to create a new endpoint for this and I was trying to get away without.